storage_controller/src/reconciler.rs

```diff
@@ -461,6 +487,39 @@ impl Reconciler {
         node: &Node,
         baseline: HashMap<TimelineId, Lsn>,
     ) -> anyhow::Result<()> {
+        // Signal to the pageserver that it should ingest up to the baseline LSNs.
+        loop {
+            match self.wait_lsn(node, tenant_shard_id, baseline.clone()).await {
+                Ok(StatusCode::OK) => {
+                    // Everything is caught up
+                    return Ok(());
+                }
+                Ok(StatusCode::ACCEPTED) => {
+                    // Some timelines are not caught up yet.
+                    // They'll be polled below.
+                    break;
+                }
+                Ok(StatusCode::NOT_FOUND) => {
+                    // None of the timelines are present on the pageserver.
+                    // This is correct if they've all been deleted, but
+                    // let let the polling loop below cross check.
+                    break;
+                }
+                Ok(status_code) => {
+                    tracing::warn!(
+                        "Unexpected status code ({status_code}) returned by wait_lsn endpoint"
+                    );
+                    break;
+                }
+                Err(e) => {
```

Comment:
This matches the old code, but maybe worth improving while we're here -- it's risky to loop indefinitely on unexpected errors.  Equally, it's risky to proceed with migration cutover if we can't reach the destination pageserver.

Maybe we need an overall threshold on time/errors here to ensure we drop out eventually in the face of errors.  Perhaps that's beyond scope of this PR though

Reply:
Can do in a follow-up. I tried to not modify external behaviour in this one.